proto/cacheclient.proto

```diff
@@ -48,3 +51,39 @@ message _SetResponse {
   ECacheResult result = 1;
   string message = 2;
 }
+
+message _HashGetRequest {
+  bytes cache_hash_name = 1;
+  bytes cache_hash_key = 2;
+}
+
+message _HashGetResponse {
+  ECacheResult result = 1;
+  bytes cache_body = 2;
+  string message = 3;
+}
+
+message _HashGetAllRequest {
+  bytes cache_hash_name = 1;
+}
+
+message HashKeyValuePair {
+  bytes key = 1;
+  bytes value = 2;
+}
+
+message _HashGetAllResponse {
+  ECacheResult result = 1;
+  repeated HashKeyValuePair cache_body = 2;
```

Comment:
I think this should be hash_body

Reply:
Using the dictionary terminology, would a better name for this just be "dictionary"?

Reply:
You'll have _name in other places so for consistency maybe use _name and _body

Reply:
Done